src/server/worker/api_server.go

```diff
@@ -141,6 +144,19 @@ func (a *APIServer) getTaggedLogger(ctx context.Context, req *ProcessRequest) (*
 			return nil, err
 		}
 		result.putObjClient = putObjClient
+		result.eg.Go(func() error {
+			for msg := range result.msgCh {
+				for _, chunk := range grpcutil.Chunk([]byte(msg), grpcutil.MaxMsgSize/2) {
+					if err := result.putObjClient.Send(&pfs.PutObjectRequest{
+						Value: chunk,
+					}); err != nil && err != io.EOF {
+						return err
+					}
+				}
+				result.objSize += int64(len(msg))
```

Comment:
I don't think so, `objSize` only gets touched here and in `Close` which first calls `eg.Wait()` so.